examples/node2vec_link.py

```diff
@@ -0,0 +1,154 @@
+import argparse
+import copy
+import os
+
+import torch
+import torch_frame
+from inferred_stypes import dataset2inferred_stypes
+from torch.utils.tensorboard import SummaryWriter
+from torch_geometric.nn import Node2Vec
+from torch_geometric.seed import seed_everything
+from torch_geometric.utils import to_undirected
+
+from relbench.data import LinkTask, RelBenchDataset
+from relbench.data.table import Table
+from relbench.data.task_base import TaskType
+from relbench.datasets import get_dataset
+from relbench.external.graph import make_pkey_fkey_graph
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--dataset", type=str, default="rel-trial")
+parser.add_argument("--task", type=str, default="rel-trial-sponsor-condition")
+parser.add_argument("--lr", type=float, default=0.001)
+parser.add_argument("--epochs", type=int, default=3000)
+parser.add_argument("--batch_size", type=int, default=512)
+parser.add_argument("--embedding_dim", type=int, default=128)
+parser.add_argument("--walk_length", type=int, default=10)
+parser.add_argument("--context_size", type=int, default=5)
+parser.add_argument("--num_workers", type=int, default=1)
+parser.add_argument("--log_dir", type=str, default="results")
+args = parser.parse_args()
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+if torch.cuda.is_available():
+    torch.set_num_threads(1)
+seed_everything(42)
+
+root_dir = "./data"
+
+# TODO: remove process=True once correct data/task is uploaded.
+dataset: RelBenchDataset = get_dataset(name=args.dataset, process=True)
+task: LinkTask = dataset.get_task(args.task, process=True)
+tune_metric = "link_prediction_map"
+assert task.task_type == TaskType.LINK_PREDICTION
+
+col_to_stype_dict = dataset2inferred_stypes[args.dataset]
+
+# We don't need to specify any text column or use any text
+# embeddings as features are not used in node2vec baseline
+for table_name, table_col_to_stype in col_to_stype_dict.items():
+    for col, stype in table_col_to_stype.items():
+        if stype == torch_frame.text_embedded:
+            col_to_stype_dict[table_name][col] = torch_frame.categorical
+
+
+data, col_stats_dict = make_pkey_fkey_graph(
+    dataset.db,
+    col_to_stype_dict=col_to_stype_dict,
+    cache_dir=os.path.join(root_dir, f"{args.dataset}_materialized_cache"),
+)
```

Comment:
These are not used, so we can just delete these lines.

Reply:
Thanks!